### PR TITLE
castero: disable more tests to fix build

### LINKED ISSUE #333019: Build failure: castero
### Steps To Reproduce

1. `nix build github:nixos/nixpkgs/master#casetro` **status:** `fail`
2. `nix build github:nixpkgs/nixos-unstable#casetro` **status:** `fail`
3. `nix build github:nixos/nixpkgs/nixos-24.05#castero` **status:** `success`
### Build log

[nix log](https://github.com/user-attachments/files/16531207/dump.txt)

### Additional context

This was discovered via pr #332628

### Notify maintainers
@TheRealKeto 

### Metadata

```console
❯  nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.10.2-cachyos, NixOS, 24.11 (Vicuna), 24.11.20240703.9f4128e`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.18.4`
 - nixpkgs: `/nix/store/dk2rpyb6ndvfbf19bkb2plcz5y3k8i5v-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by TheRealKeto**
## Description of changes

Closes #333019 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by JeremiahSecrist** [comment]
looks like `test_display_get_y_n` also failed when I ran the pr check.

File: pkgs/by-name/ca/castero/package.nix [hunk 1]
```diff
@@ -39,7 +39,11 @@ python3.pkgs.buildPythonApplication rec {
     "tests"
   ];
 
-  disabledTests = [ "test_datafile_download" ];
+  disabledTests = [
```
Comment (AndersonTorres) [inline pkgs/by-name/ca/castero/package.nix:42]: When possible, put comments describing why the tests were disabled.

Optionally, out of courtesy, inform upstream about the errors.

**Review by AndersonTorres — COMMENTED** [review]
Besides LGTM

File: pkgs/by-name/ca/castero/package.nix [hunk 2]
```diff
@@ -39,7 +39,11 @@ python3.pkgs.buildPythonApplication rec {
...
+    "test_datafile_download"
+    "test_display_get_input_str"
+    "test_display_y_n"
```
Comment (JeremiahSecrist) [inline pkgs/by-name/ca/castero/package.nix:45]:
```suggestion
    "test_display_y_n"
    "test_display_get_y_n"
```

Comment (TheRealKeto) [reply pkgs/by-name/ca/castero/package.nix]: This was a typo, fixed in the latest commit. Sorry for any delays.

**Review by JeremiahSecrist — APPROVED** [review]
LGTM:
Result of `nixpkgs-review pr 333055` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>castero</li>
    <li>castero.dist</li>
  </ul>
</details>

File: pkgs/by-name/ca/castero/package.nix (hunk 2 above)
Comment (JeremiahSecrist) [reply pkgs/by-name/ca/castero/package.nix]: Thank you so much for your time!

**Pull request merged**